website/docs/concepts/provider_lifecycles.mdx

```diff
@@ -0,0 +1,87 @@
+---
+title: Provider Lifecycles
+---
+
+## When does my Provider get created and disposed?
+
+The states that all different types of providers can go through are the same:
+
+- Uninitialized
+- Creating
+- Alive
+- Disposing
+
+### Uninitialized
+
+An uninitialized provider does not take up any memory since its state is not initialized.
+It will stay that way until an Alive provider or the UI reads or watches it.
+
+### Creating -> Alive
+
+When an uninitialized provider is being read or watched it will be created.
```

Comment:
The provider isn't created. Its state is

I don't think it's necessary to mention "read or watched". It sounds redundant and ignores ref.listen